### PR TITLE
Custom functions

### PR DESCRIPTION
**Pull request by gzsombor**
Add ability to parse custom functions - without explicitly needed to add each function to the parser

### CODE REVIEW

**Comment by ms705** [comment]
Thank you! This looks interesting, and is something I've been wanting to add a for a while. Let me take a read of the changes over the weekend; on the surface, this looks great :+1:

File: src/column.rs [hunk 1]
```diff
@@ -33,21 +34,48 @@ impl Display for FunctionExpression {
...
 #[derive(Clone, Debug, Eq, Hash, PartialEq, Serialize, Deserialize)]
-pub enum FunctionArguments {
+pub struct FunctionArguments {
+    pub arguments: Vec<FunctionArgument>,
+}
+
+impl Display for FunctionArguments {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let mut first = true;
+        for argument in self.arguments.iter() {
```
Comment (ms705) [inline src/column.rs:50]: You can use https://doc.rust-lang.org/stable/std/primitive.slice.html#method.join here to simplify this code

**Review by ms705 — APPROVED** [review]
LGTM, see comment on replacing the argument print loop with a `.join()` though.

File: src/column.rs (hunk 1 above)
Comment (gzsombor) [reply src/column.rs]: Done

**Comment by gzsombor** [comment]
I've fixed the test too - I don't know how could I wrote that test et all. Probably, when I separated my changes into 3 PR, I wasn't thorough enough.

**Review by ms705 — APPROVED** [review]

**Comment by ms705** [comment]
Thank you! :+1:

**Pull request merged**